cli/src/commands/uninstall.rs

```diff
@@ -0,0 +1,140 @@
+//! Phylum CLI removal.
+
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+use anyhow::{anyhow, Result};
+use clap::ArgMatches;
+use tempfile::NamedTempFile;
+
+use crate::commands::{CommandResult, ExitCode};
+use crate::{config, print_user_success, print_user_warning};
+
+/// Handle the `uninstall` subcommand.
+pub fn handle_uninstall(matches: &ArgMatches) -> CommandResult {
+    let home_dir = home_dir()?;
+
+    if matches.is_present("purge") {
+        purge(&home_dir);
+    }
+
+    remove_installed_files(&home_dir);
+
+    if let Err(err) = Shell::Bash.cleanup(&home_dir) {
+        print_user_warning!("BASH config error: {}", err);
+    }
+
+    if let Err(err) = Shell::Zsh.cleanup(&home_dir) {
+        print_user_warning!("ZSH config error: {}", err);
+    }
+
+    Ok(ExitCode::Ok.into())
+}
+
+/// Remove the entire `~/.config/phylum` directory.
+fn purge(home_dir: &Path) {
+    let config_dir = config::config_dir(home_dir).join("phylum");
+
+    match fs::remove_dir_all(&config_dir) {
+        Ok(()) => print_user_success!("Successfully removed {:?}", config_dir),
+        Err(err) => {
+            print_user_warning!("Could not remove phylum config directory: {}", err);
+        }
+    }
+}
+
+/// Remove files created by `install.sh`.
+fn remove_installed_files(home_dir: &Path) {
+    let data_dir = data_dir(home_dir).join("phylum");
+    let bin_path = bin_dir(home_dir).join("phylum");
+
+    let data_result = fs::remove_dir_all(data_dir);
+    let bin_result = fs::remove_file(bin_path);
```

Comment:
Some day, we'll have to come up with a solution that works on Windows where you can't just `unlink` the running exectuable. Lucky for us, today is not that day.